src/main/java/net/minecraftforge/common/ForgeMod.java

```diff
@@ -141,6 +144,22 @@ public void serverStarting(FMLServerStartingEvent evt)
     {
         new ForgeCommand(evt.getCommandDispatcher());
         ConfigCommand.register(evt.getCommandDispatcher());
+        evt.getServer().getResourceManager().addReloadListener(new ReloadListener<Object>()
```

Comment:
Why is this needed if we get the reload event?

Reply:
The TagsUpdatedEvent is only posted on the client when the sync packet is received.  The server needs to update on data reload.

Reply:
Seems like a lack in the update event.
I'd have to look into it to be sure, but I think reload listener should be fired on both client and server in this context.